rust/connlib/snownet/src/node.rs

```diff
@@ -440,6 +448,35 @@ where
         self.buffered_transmits.pop_front()
     }
 
+    pub fn upsert_turn_servers(
+        &mut self,
+        servers: &HashSet<(RId, SocketAddr, String, String, String)>,
```

Comment:
Maybe we can use the opportunity in this PR to use a struct here instead of a tuple?

Reply:
I couldn't be asked :sweat_smile: 

Maybe once we remove the STUN-only stuff and things settle a bit around this part of the codebase.